enumerables.rb

```diff
@@ -0,0 +1,127 @@
+# rubocop: disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n])
+      n += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n], n)
+      n += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |n|
+      result.push(n) if yield(n)
+    end
+    result
+  end
+
+  def my_all?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        result = false unless yield(n)
+      elsif var.nil?
+        result = false unless n
+      else
+        result = false unless var == n
+      end
+    end
+    result
+  end
+
+  def my_any?(var = nil)
+    result = false
+    my_each do |n|
+      if block_given?
+        return result = true if yield(n)
+      elsif var.nil?
+        return result = true unless n
+      else
+        return result = true unless var == n
+      end
+    end
+    result
+  end
+
+  def my_none?(var = nil)
+    result = true
+    my_each do |n|
+      if block_given?
+        return result = false if yield(n)
+      elsif var.nil?
+        return result = false unless n
+      else
+        return result = false unless var == n
+      end
+    end
+    result
+  end
```

Comment:
* [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
  ```ruby
    array = [false,nil, [], false]
    p array.my_none? == array.none? # false
  ```
* [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
  ```ruby
    array = [1,2,5,4,7]
    p array.my_none?(String) == array.none?(String) # false
  ```
* [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
  ```ruby
    array = %w[dog door rod blade]
    p array.my_none?(/c/) == array.none?(/c/) # false
  ```
* [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
  ```ruby
    array = %w[dog door rod blade]
    p array.my_none?(5) == array.none?(5) # false
  ```

Reply:
Same mistake as above.